File: Python/Practice/LeetCode/Easy/Count of Matches in Tournament.py
# https://leetcode.com/problems/count-of-matches-in-tournament/description/

class Solution:
    def numberOfMatches(self, n: int) -> int:
        numberOfTeams = n
        matchesCounter = 0

        while (numberOfTeams != 1):
            matchesCounter += math.floor(numberOfTeams / 2)
            numberOfTeams = math.ceil(numberOfTeams / 2)
        
        return matchesCounter;

File: Java/Practice/LeetCode/Easy/Count of Matches in Tournament.java
// https://leetcode.com/problems/count-of-matches-in-tournament/description/

class Solution {
    public int numberOfMatches(int n) {
        int numberOfTeams = n;
        int matchesCounter = 0;

        while (numberOfTeams != 1) {
            matchesCounter += Math.floor(numberOfTeams / 2);
            numberOfTeams = (int) Math.ceil((double) numberOfTeams / 2);
        }
        
        return matchesCounter;
    }
}